src/System.Private.ServiceModel/src/System/ServiceModel/Channels/MessageContent.cs

```diff
@@ -186,7 +186,7 @@ protected override Task<Stream> CreateContentReadStreamAsync()
 
         protected override Task SerializeToStreamAsync(Stream stream, TransportContext context)
         {
-            return _messageEncoder.WriteMessageAsync(_message, _stream);
+            return _messageEncoder.WriteMessageAsync(_message, stream);
```

Comment:
There are two ways that HttpClient can call into an HttpContent class. One is to call CreateContentReadStreamAsync, and the other is to call SerializeToStreamAsync. The first way expects a stream to be returned that the content can be read from. The second passes in a stream that the content should be written to so doesn't use the _stream field. The .Net Native implementation of HttpClient uses the first method while the coreclr implementation based on WinHTTP uses the second method.